python/ray/util/spark/start_ray_node.py

```diff
@@ -62,7 +62,10 @@
     # same temp directory, adding a shared lock representing current ray node is
     # using the temp directory.
     fcntl.flock(lock_fd, fcntl.LOCK_SH)
-    process = subprocess.Popen([ray_cli_cmd, "start", *arg_list], text=True)
+    process = subprocess.Popen(
+        [sys.executable, shutil.which(ray_cli_cmd), "start", *arg_list],
```

Comment:
this will become `python /path/to/ray start ....`?

Reply:
yes.

Reply:
and it ensures the `python` is the one that runs `start_ray_node`

Reply:
In contrast,
`ray start ..` command uses python that is set by Shebang `!# ...`, the Shebang line is hardcoded in `ray` script, it can't be changed to other python executable path.

Reply:
`the Shebang line is hardcoded in ray script,` do you know where is it?

Reply:
Also let's turn what you said above to comments.

Reply:
@jjyao 
The shebang is generated by `pip install ` , e.g., if you run `cat `which ray`, you will get

```
(py38) weichen.xu@C02G36DSMD6R /tmp % cat `which ray`
#!/Users/weichen.xu/opt/miniconda3/envs/py38/bin/python
# -*- coding: utf-8 -*-
import re
import sys
from ray.scripts.scripts import main
if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
```

this line `#!/Users/weichen.xu/opt/miniconda3/envs/py38/bin/python` is hardcoded